test/e2e/grpc_test.go

```diff
@@ -64,30 +64,30 @@ func hasPort(u string) bool {
 	return err == nil
 }
 
-func dial(host, domain string) (*grpc.ClientConn, error) {
+func dial(ctx *TestContext, host, domain string) (*grpc.ClientConn, error) {
+	defaultPort := "80"
+	if test.ServingFlags.HTTPS {
+		defaultPort = "443"
+	}
 	if !hasPort(host) {
 		host = net.JoinHostPort(host, defaultPort)
 	}
 	if !hasPort(domain) {
 		domain = net.JoinHostPort(domain, defaultPort)
 	}
 
-	if host != domain {
-		// The host to connect and the domain accepted differ.
-		// We need to do grpc.WithAuthority(...) here.
-		return grpc.Dial(
-			host,
-			grpc.WithAuthority(domain),
-			grpc.WithInsecure(),
-			// Retrying DNS errors to avoid .xip.io issues.
-			grpc.WithDefaultCallOptions(grpc.WaitForReady(true)),
-		)
+	secureOpt := grpc.WithInsecure()
+	if test.ServingFlags.HTTPS {
+		tlsConfig := test.TLSClientConfig(context.Background(), ctx.t.Logf, ctx.clients)
+		tlsConfig.ServerName = strings.Split(domain, ":")[0] // Set ServerName for pseudo hostname with TLS.
```

Comment:
Should we use `net.SplitHostAndPort` instead? Probably doesn't really make a difference and we can safely drop error handling here anyway I think.